tendermint/src/block/commit_sig.rs

```diff
@@ -74,47 +77,61 @@ impl TryFrom<RawCommitSig> for CommitSig {
                 let timestamp = value.timestamp.unwrap();
                 // 0001-01-01T00:00:00.000Z translates to EPOCH-62135596800 seconds
                 if timestamp.nanos != 0 || timestamp.seconds != -62135596800 {
-                    return Err(Kind::InvalidTimestamp
-                        .context("absent commitsig has non-zero timestamp")
-                        .into());
+                    return Err(error::invalid_timestamp_error(anyhow::anyhow!(
+                        "absent commitsig has non-zero timestamp"
+                    )));
                 }
             }
             if !value.signature.is_empty() {
-                return Err(Kind::InvalidSignature.into());
+                return Err(error::invalid_signature_error(anyhow::anyhow!(
+                    "invalid signature error"
+                )));
             }
             return Ok(CommitSig::BlockIdFlagAbsent);
         }
         if value.block_id_flag == BlockIdFlag::Commit.to_i32().unwrap() {
             if value.signature.is_empty() {
-                return Err(Kind::InvalidSignature
-                    .context("regular commitsig has no signature")
-                    .into());
+                return Err(error::invalid_signature_error(anyhow::anyhow!(
+                    "regular commitsig has no signature"
+                )));
             }
             if value.validator_address.is_empty() {
-                return Err(Kind::InvalidValidatorAddress.into());
+                return Err(error::invalid_validator_address_error());
             }
             return Ok(CommitSig::BlockIdFlagCommit {
                 validator_address: value.validator_address.try_into()?,
-                timestamp: value.timestamp.ok_or(Kind::NoTimestamp)?.try_into()?,
+                timestamp: value
+                    .timestamp
+                    .ok_or(error::no_timestamp_error())?
+                    .try_into()
+                    .map_err(|e: std::convert::Infallible| {
```

Comment:
`Infallible` can never happen. So you can eliminate it with just `match e {}`

Reply:
This is still not fixed. The `Infallible` variant can also be removed.

Reply:
You'll report an error if you don't clearly identify it.